modules/service/src/main/scala/lucuma/odb/service/SequenceService.scala

```diff
@@ -568,8 +529,47 @@ object SequenceService {
           $step_config_smart_gcal
       """.command
 
-    val SelectStepConfigSmartGcalForObs: Query[Observation.Id, (Step.Id, StepConfig.SmartGcal)] =
-      selectStepConfigForObs("t_step_config_smart_gcal", StepConfigSmartGcalColumns, step_config_smart_gcal)
+    private val step_config: Codec[StepConfig] =
+      (
+        step_type               *:
+        step_config_gcal.opt    *:
+        step_config_science.opt *:
+        step_config_smart_gcal.opt
+      ).eimap { case stepType *: oGcal *: oScience *: oSmart *: EmptyTuple =>
```

Comment:
It seems pointlessly confusing that for codecs `a *: b *: c` is the only way to write it, but for tuples (which end in `EmptyTuple`) you can also write it as `(a, b, c)`.

![image](https://github.com/gemini-hlsw/lucuma-odb/assets/1200131/830470ea-bac2-42a3-afe5-9521691a2cc7)

Reply:
Thanks for the tip, that's much better.